modules/jpa/src/main/java/io/oasp/module/jpa/dataaccess/api/QueryDslHelper.java

```diff
@@ -0,0 +1,512 @@
+package io.oasp.module.jpa.dataaccess.api;
```

Comment:
I think we should extract the search pattern query support to a seperate util to also document in the class javadocs, that this is not for general use, but the newXClause functions come with a special use case.
Otherwise, you will end up in users writing their queries just which this provided functions, which would end up in a code mess, which cannot be understood easily as you cannot trace the query construction anymore.

Reply:
Sorry, but I do not really get this point. Should the `QueryDslHelper` class go to a different package, a different module (and package) or should it be split into multiple classes? Can you give a more concrete suggestion how to improve?

Reply:
The `QueryDslHelper` contains more than just the newXClause methods. It also contains the pagination creation methods. The latter once might be access by programmers frequently. The newXClause methods however should not as they follow a very specific use case. So I would simply differentiate between both functionalities by seperating them in two different util classes.

Reply:
Should then the separate class holding the newXClause methods extend the other class? Otherwise writing repositories gets unnecessary more complicated as you have to know where to find which method. I will give it a try this way. However, then one can still use the extended class in the same way as the current class `QueryDslHelper` as it is now anywhere. I know that separating aspects is a good idea in general. But this will also raise naming issues and might even lead to more complexity and confusion as the developers using this have to understand what to use when and how the relation between these separated classes is actually designed.
Currently `QueryDslHelper` is quite obvious: You get a lot of utility/helper methods that help you with QueryDsl. How should we name the extracted class?

Reply:
>  However, then one can still use the extended class in the same way as the current class QueryDslHelper as it is now anywhere.

yes, maybe we are talking about a matter of taste. I just wanted to point out, that there is a certain probability of misuse of the newXClause. But as long as we want keeping inheritance as the way to probagate common API, there is no way out. I would have extracted this to a static util class as this functionality serves a very specific need, i.e. supporting advanced query syntax entered by a user. So why not point our users, who really need it to the util class?
Would even make the API a little bit clearer as the general context is defined by the util class. By using static imports, you even do not produce more unreadable in your dao methods.